src/common/snippets/src/lowered/pass/extract_loop_invariants.cpp

```diff
@@ -0,0 +1,199 @@
+                                         LinearIR::constExprIt& inner_loop_begin_pos,
+                                         LinearIR::constExprIt& inner_loop_end_pos) {
+    // update expr loop id
+    remove_last_loop_id(expr);
+    // move if it is not the first
+    if (expr != *inner_loop_begin_pos) {
+        auto port_expr_iter = std::find(inner_loop_begin_pos, inner_loop_end_pos, expr);
+        OPENVINO_ASSERT(port_expr_iter != inner_loop_end_pos, "Identified extractable expr is not found in loop.");
+        linear_ir.move(port_expr_iter, inner_loop_begin_pos);
+    } else {
+        inner_loop_begin_pos++;
+    }
+}
+
+void ExtractLoopInvariants::update_loop_ports(const ExpressionPtr& expr, const LoopManagerPtr& loop_manager, size_t inner_loop_id,
+                                              LinearIR::constExprIt& inner_loop_begin_pos, LinearIR::constExprIt& inner_loop_end_pos) {
+    const auto& inner_loop_info = loop_manager->get_loop_info<UnifiedLoopInfo>(inner_loop_id);
+    // delete expr input ports from loop input points, add expr output ports' consumers if
+    // consumed in inner loop to loop input ports.
+    std::vector<ExpressionPort> new_loop_input_ports;
+    for (size_t i = 0; i < expr->get_output_count(); i++) {
+        const auto& consumers = expr->get_output_port_connector(i)->get_consumers();
+        for (const auto& consumer : consumers) {
+            const auto& loop_ids = consumer.get_expr()->get_loop_ids();
+            if (std::find(loop_ids.cbegin(), loop_ids.cend(), inner_loop_id) != loop_ids.cend()) {
+                new_loop_input_ports.push_back(consumer);
+            }
+        }
+    }
+    const auto& expr_input_ports = expr->get_input_ports();
+    inner_loop_info->update_loop_ports(expr_input_ports, new_loop_input_ports, true);
+
+    // delete expr out ports from loop out ports directly if it's in loop output ports
+    const auto& loop_out_ports = inner_loop_info->get_output_ports();
+    std::vector<ExpressionPort> exp_out_ports;
+    for (size_t i = 0; i < expr->get_output_count(); ++i) {
+        const auto& out_port = expr->get_output_port(i);
+        if (inner_loop_info->find_loop_port(out_port) != loop_out_ports.end()) {
+            exp_out_ports.push_back(out_port);
+        }
+    }
+    if (!exp_out_ports.empty()) {
+        std::vector<ExpressionPort> new_ports;
+        inner_loop_info->update_loop_ports(exp_out_ports, new_ports, false);
+    }
+    // need sort after update loop ports. There are possibility that all exprs are moved to outer loop.
+    if (!inner_loop_info->get_input_ports().empty() && !inner_loop_info->get_output_ports().empty()) {
+        loop_manager->sort_loop_ports(inner_loop_begin_pos, inner_loop_end_pos, inner_loop_id);
+    }
+}
+
+bool ExtractLoopInvariants::extract_from_loop(const size_t& inner_loop_id, LinearIR& linear_ir) {
+    const auto& loop_manager = linear_ir.get_loop_manager();
+    bool status = false;
```

Comment:
They could be different values as `has_been_extraction ` is reset in a new iteration. It will be false in the last iteration, means no more expr is available to extract in the loop and then finish for this loop. `status `will update to true when a expr is extracted in any iteration.